### PR TITLE
feat(device-discovery): discover StackWise Virtual switch stacks

### LINKED ISSUE #498: Feature Request - ORB discovery with Stackwise-Virtual Switch Stacks
[Issue 251](https://github.com/netboxlabs/orb-agent/issues/251) allows for dedicated Stackwise switches to be detected, but fails on Stackwise-Virtual devices such as Cisco 9400/9500/9500 devices.

The issue appears to be related to the "show switch detail" command, which does not exist on the latter devices, and is instead "show switch" on the virtually stacked devices. I believe all other logic is identical and should work appropriately.

This feature would come with the same benefits as Issue 251 for additional devices. As with before, it is a single ssh connection for the virtual stack.

### PR DESCRIPTION
**Pull request by leoparente**
Closes #498.

## What

Cisco Catalyst 9400 / 9500 / 9600 in **StackWise Virtual** (SVL) were not discovered as a NetBox `VirtualChassis`. The pair landed as two unrelated `Device` records, with interfaces and IPs split across them arbitrarily.

The reporter's diagnosis was that `show switch detail` does not exist on these platforms and `show switch` does. That is true, but it is not the whole fix, and the interesting part is what it does not cover.

## Why the command swap alone is not enough

`show switch` and `show switch detail` resolve to the **same** ntc-template via the index regex `sh[[ow]] sw[[itch]]( d[[etail]])?\s*$`. Verified: both produce byte-identical parses. So changing the command string changes no parsing behaviour at all.

The real blocker was **serials**. A member is dropped unless we can find its serial, and serials come from `show inventory`. The matcher accepted a NAME of exactly `Switch 1`; on SVL the chassis row is named `Switch 1 Chassis`, so nothing matched, every member was dropped, and discovery emitted nothing. Fixing the command without fixing this would have looked correct and still produced no virtual chassis.

## The trap in the obvious inventory fix

The tempting change is to let any suffix follow the number, `^Switch\s+(\d+)\b`. That also matches `Switch 1 Power Supply Module 0`, `Switch 1 Fan Tray 0` and `Switch 1 Slot 1 Supervisor`, and because the index was last-write-wins the member's identity came from whichever component row appeared last.

On the reporter's C9500 that accidentally yields the right answer, because the chassis and supervisor report the same PID and serial. On a modular 9400, where the supervisor has its own PID, the same code was measured to produce:

```
member 1  model = C9400-PWR-3200AC     <- the power supply
          serial = the power supply's serial
```

**Serial is NetBox's device matcher**, so that attaches discovery to the wrong device record. The matcher stays anchored, and a test pins the modular case so the anchor cannot be "simplified" back out.

Serial and model are now also committed from the **same** inventory row. They were written through two independent conditionals, so a bare `Switch N` row and a `Switch N Chassis` row could each supply one field.

## Incidental fix: degraded physical stacks emitted nothing

While replacing the table parse I found a live bug on the path that already works. `cisco_ios_show_switch_detail.textfsm` declares `Value STATE (\w+)`, so a multi-word state raises `TextFSMError` — and because the template is all-or-nothing, **every** member row is discarded, not just the offending one.

| Member state | Template result |
|---|---|
| `Ready` | 2 rows |
| `Provisioned` | 2 rows |
| `Version Mismatch` | **TextFSMError — all rows lost** |
| `Sync not started` | **TextFSMError — all rows lost** |

So a 9300 stack with one member mid-upgrade produced no `VirtualChassis` at all, silently. A driver-local per-row parser replaces the template for this one table; it degrades one row at a time instead. A fixture pins the case.

Parity is proven, not asserted: for all 8 pre-existing fixtures the local parser agrees with the template on **every** shared field, and where the template raises it must recover the row count recorded in `_TEMPLATE_RAISES_EXPECTED_ROWS` — a new raising fixture fails loudly until its expectation is written down.

## Domain number

`show stackwise-virtual` is the only command that exposes the SVL domain, and it has no upstream template, so one regex reads `Domain Number : N`. SVL stacks therefore carry a domain that physical stacks cannot supply. It is purely additive: any failure leaves the domain `None`, which is what physical stacks have always emitted.

`show switch virtual` is **not** used — it does not exist on these platforms. An upstream template for it does exist, which is exactly what made it look viable.

## Observability

The reported symptom was silence, so the failure paths are now distinguishable:

- A CLI rejection on every attempted command, or an explicit standalone answer → **DEBUG**. This *removes* a warning that every non-Catalyst IOS device (ISR / ASR / CSR) and every standalone Catalyst emitted on **every discovery cycle**, since `show switch detail` raises `TextFSMError` on them today.
- Output that looks like a member table but yields no rows → **WARNING**, naming the host and the commands tried. This is the diagnosable case and the one the next unsupported platform lands in.
- Fewer than two members → **WARNING** with the count, in the driver and again at `validate_chassis_payload`, so a partial parse degrading to a single `Device` is visible.
- Empty output classifies as uninformative rather than surprising, so a silent command cannot force a warning.

Every message now carries the hostname. None did before, which made a fleet warning unattributable.

## Evidence

Fixtures are verbatim captures from a real C9500-48Y4C SVL pair on IOS-XE 17.15/17.18, supplied in #498: the `show switch detail` rejection, the `show switch` table, `show stackwise-virtual`, and `show inventory`.

## Verification

- `1957 passed, 161 skipped`; `ruff check .` clean; 4 Go modules 0 issues.
- Mutation-checked rather than trusted: reverting the ` Chassis` suffix fails 6 tests; removing the `show switch` fallback fails 2, **including** the end-to-end SVL fixture. Those `expected_result.json` files were generated from driver output, so this was worth proving.
- Adversarial edge cases: 2-digit ids, `Removed/Absent`, tabs and trailing whitespace all parse; parenthesised states, blank priority, separator-less MACs, 3-digit ids and the Stack Port / **Neighbors** data rows are all rejected. Every rejection skips a row rather than producing a wrong one.

## Accepted limitations

- `cisco_ios_show_inventory.textfsm` captures `SN` as `([\w+\d+]+)`, so a serial containing a character outside `[A-Za-z0-9_+]` truncates silently. Cisco serials are alphanumeric; the fix is upstream.
- DEBUG is currently invisible in device-discovery — the root logger is pinned to INFO by an import-time `basicConfig` and there is no log-level flag. The DEBUG paths above are therefore deliberately silent, not diagnostic. A `--log-level` flag is specified separately.
- A row with no version column **and** a multi-word state would mis-split. No such combination has been observed; every multi-word state occurs on a present member, which reports a version.

## Draft

Draft while a broader adversarial review finishes. Behaviour, tests and docs are complete; anything that review confirms will land as additional commits before this is marked ready.

---

## Adversarial review pass

A whole-branch adversarial review (5 lenses, each finding independently refuted before it counted) raised 35 findings; **7 survived refutation**. All 7 are addressed in `9bb8acc`. They clustered into two themes, and both were mine.

**The new diagnostics fired on healthy hardware.** A standalone stack-capable Catalyst (2960S/2960X/3850/9200/9300) answers `show switch detail` with a one-row member table — the most common deployment in any fleet. That produced **two** WARNINGs per discovery cycle, one from the driver and one from `validate_chassis_payload`, for a device behaving exactly as expected. Neither warning existed before this branch, so it was a self-inflicted noise regression that directly contradicted the noise policy this same branch states.

Reproduced against ntc-templates' own real capture, then fixed by warning only when members were **lost**:

| Scenario | Before | After |
|---|---|---|
| Standalone stack-capable Catalyst | 2 WARN | **0** (DEBUG) |
| Healthy 2-member stack | 0 | 0 |
| Real stack, one serial unresolvable | 2 WARN | 2 WARN (correctly — this one is a real gap) |

`validate_chassis_payload` is driver-agnostic and a single-member payload is documented-normal for several drivers, so it now warns only when validation actually dropped members.

**Two branches were dead to the test suite.** Mutation-checked, not assumed:

- Mutating `looked_like_a_table` to `False` broke **no** test. On inspection it was redundant — output that looks like a table but parses to nothing already reaches the WARNING branch via `not rejected` — and actively harmful, since it would misfire on a release printing the column header above an explicit `Switch is not on any stack.`, turning a standalone device into a warning. **Deleted rather than tested.**
- Replacing the `send_command`-exception `continue` with `return None` broke **no** test. That early return would make SVL discovery depend on a command those platforms do not implement. Now pinned by a test that fails under exactly that mutation.
- The `translate_chassis` guard had zero coverage; it now has two tests plus a mutation check on the `members_raw` gate.

Final: **1961 passed, 161 skipped**, ruff clean.

### CODE REVIEW

**Comment by leoparente** [comment]
@codex review

File: orb-discovery/device-discovery/custom_napalm/ios.py [hunk 1]
```diff
@@ -266,53 +266,260 @@ def classify_module_type_cisco_ios(pid: str) -> _ModuleType:
...
+# because the chassis and supervisor report the same PID and serial, but on a
+# modular 9400 it yields model="C9400-PWR-3200AC" and the power supply's
+# serial. Serial is NetBox's device matcher, so that attaches discovery to the
+# WRONG device record. Do not relax the trailing anchor.
+#
+# A bare "Chassis" (standalone, no member id) deliberately does not match — the
+# caller treats an empty index as "no per-member inventory available" and the
+# affected members are dropped by to_payload().
+_INVENTORY_CHASSIS_RE = re.compile(
+    r"^\s*(?:Switch\s+(\d+)(?:\s+Chassis)?|(\d+))\s*$",
```
[the remark this answers is not in the document]
Comment (leoparente) [reply orb-discovery/device-discovery/custom_napalm/ios.py]: Valid, fixed in 655bc34 (plus a follow-up making the new test's docstring raw for ruff D301).

This was an internal inconsistency in my own change rather than a judgement call, which is what makes it a good catch. The same file already accommodates the no-space form in three places and I did not match it:

```
ios.py:620  _INVENTORY_VC_SLOT_RE  r"^Switch\s*(\d+)\s+Slot\s+(\d+)..."
ios.py:630  _INVENTORY_VC_FRU_RE   r"^Switch\s*(\d+)\s+FRU\s+Uplink\s+Module\s+(\d+)"
ios.py:689  _SWITCH_PREFIX_RE      r"^Switch\s*(\d+)\b"
```

All three use `\s*`; my new chassis matcher used `\s+`. Confirmed the consequence you describe, before fixing:

```
'Switch 1 Chassis'  -> MATCH    {1: 'CAT11111111'}
'Switch1 Chassis'   -> no match {}
'Switch1'           -> no match {}
```

So on a release emitting the no-space form, `serial_by_id` stays empty, `to_payload` drops every member, and the SVL pair still falls back to a single Device — the exact failure this branch exists to remove, just narrowed to a subset of releases.

Fixed as you suggested: `\s*` before the id, trailing anchor untouched. I kept the space before `Chassis` mandatory rather than also relaxing it, since no observed capture omits it and relaxing both would widen the match for no evidenced gain.

The component-row protection is unaffected and now has a test of its own — `Switch1 Power Supply Module 0`, `Switch1 Slot 1 Supervisor` and `Switch1 Fan Tray 0` all still yield nothing, which matters because that anchor is what stops a member taking a power supply's serial on a modular 9400.

Mutation-checked: reverting to `\s+` fails `test_accepts_the_no_space_switch_form` and `test_no_space_bare_switch_form_also_accepted`.

Full suite 1964 passed, 161 skipped; ruff clean; the four Go modules report 0 issues.

**Comment by leoparente** [comment]
@codex review

**Review by samiura — APPROVED** [review]

**Review by jajeffries — APPROVED** [review]
